lib/util/clone.ts

```diff
@@ -1,5 +1,33 @@
-import safeStringify from 'fast-safe-stringify';
+import is from '@sindresorhus/is';
 
-export function clone<T>(input: T = null): T {
-  return JSON.parse(safeStringify(input));
+export function clone<T>(input: T, refs = new WeakSet<any>()): T {
+  if (is.array(input)) {
+    refs.add(input);
+    const output: unknown = new Array(input.length);
+    for (let idx = 0; idx < input.length; idx += 1) {
+      const elem = input[idx];
+
+      if (refs.has(elem)) {
+        return undefined;
+      }
+
+      output[idx] = clone(elem, refs);
+    }
+    refs.delete(input);
```

Comment:
I think we don't need this
```suggestion
```

Reply:
It's needed for tracking descendants, while siblings can reference to the same object. Though, we won't maintain this same-ness and just clone them again and again.

Reply:
(See also one of the tests)

Reply:
Is this same behavior as before? So we should at least maintain behavior.

I also like to get rid of the undefined properties, as they consume memory and our config objects are already pretty big.

Reply:
I would say it's pretty same. The current one replaces any value with `'[Circular]` string. Maybe, we can silently ignore it. Also, I remove the entire array containing circular item, because processing of such an items probably can be unexpected, while removing just the one item can be incorrect too.